resolver/ecs_resolver.go

```diff
@@ -0,0 +1,104 @@
+package resolver
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/0xERR0R/blocky/config"
+	"github.com/0xERR0R/blocky/model"
+	"github.com/0xERR0R/blocky/util"
+	"github.com/miekg/dns"
+)
+
+// https://www.rfc-editor.org/rfc/rfc7871.html#section-6
+const (
+	ecsSourceScope = uint8(0)
+
+	ecsMaskIPv4 = uint8(net.IPv4len * 8)
+	ecsMaskIPv6 = uint8(net.IPv6len * 8)
+)
+
+// https://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml
+const (
+	ecsFamilyIPv4 = uint16(iota + 1)
+	ecsFamilyIPv6
+)
+
+// ECSMask is an interface for all ECS subnet masks as type constraint for generics
+type ECSMask interface {
+	config.ECSv4Mask | config.ECSv6Mask
+}
+
+// A EcsResolver is responsible for adding the subnet information as EDNS0 option
+type EcsResolver struct {
```

Comment:
Had to go in the RFC to figure out what `ECS` means. Let's add it to the comment:
```suggestion
// EcsResolver is responsible for adding the EDNS Client Subnet information as EDNS0 option.
type EcsResolver struct {
```

And if possible rename to have `Ecs` in all caps since it's an abbreviation. The `New` function too in that case.